aspnetcore/blazor/fundamentals/dependency-injection.md

```diff
@@ -961,6 +961,111 @@ The app can register transient disposables without throwing an exception. Howeve
 Navigate to the `TransientExample` component at `/transient-example` and an <xref:System.InvalidOperationException> is thrown when the framework attempts to construct an instance of `TransientDependency`:
 
 > System.InvalidOperationException: Trying to resolve transient disposable service TransientDependency in the wrong scope. Use an 'OwningComponentBase\<T>' component base class for the service 'T' you are trying to resolve.
+  
+## Access Blazor services from a different DI scope
+  
+*This section only applies to Blazor Server apps.**
+
+There may be times when a Razor component invokes asynchronous methods that execute code in a different DI scope. Without the correct approach, these DI scopes don't have access to Blazor's services, such as <xref:Microsoft.JSInterop.IJSRuntime> and <xref:Microsoft.AspNetCore.Components.Server.ProtectedBrowserStorage>.
+
+For example, <xref:System.Net.Http.HttpClient> instances created using <xref:System.Net.Http.IHttpClientFactory> have their own DI service scope. As a result, <xref:System.Net.Http.HttpMessageHandler> instances configured on the <xref:System.Net.Http.HttpClient> aren't able to directly inject Blazor services.
+
+Until this scenario is addressed as a product feature in a future Blazor release, the following workaround is available to configure accessible Blazor services from code in other DI scopes when invoked from a Razor component.
+
+Create a static class `BlazorServiceAccessor` that defines an [`AsyncLocal<IServiceProvider>`](xref:System.Threading.AsyncLocal`1), which stores the Blazor <xref:System.IServiceProvider> for the current asynchronous context. Asynchronous code invoked from a Razor component can use `BlazorServiceAccessor.Services` to access Blazor services. *This is only necessary if the <xref:System.IServiceProvider> available to the invoked code was created in a different DI service scope.*
+
+`BlazorServiceAccessor.cs`:
+
+```csharp
+internal static class BlazorServiceAccessor
+{
+    private static readonly AsyncLocal<IServiceProvider> s_blazorServices = new();
+
+    public static IServiceProvider Services
+    {
+        get => s_blazorServices.Value ?? throw new InvalidOperationException(
+            "Blazor services are not available in the current context.");
+        set => s_blazorServices.Value = value;
+    }
+}
```

Comment:
Let's follow the pattern that we use in other places. Register this as a singleton service in DI and have the component class inject this type (like HttpContextAccessor).